File: Data Structures & Algorithms/two-integer-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const map = {};

        for (let i = 0; i < nums.length; i++) {
            const n = target - nums[i];

            if (n in map) {
                return [i, map[n]];
            } else {
                map[nums[i]] = i;
            }
        }

        return [-1, -1];
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l, r = 0, len(nums) - 1

        while l < r:
            left = nums[l]
            right = nums[r]

            s = left + right

            if s > target:
                r = r - 1
            elif s < target:
                l = l + 1
            else:
                return [l + 1, r + 1]
        
        return [-1, -1]